stdlib/ballerina-jms/src/main/ballerina/jms/connection.bal

```diff
@@ -46,11 +46,15 @@ documentation { Configurations related to a JMS connection
     F{{initialContextFactory}} JMS provider specific inital context factory
     F{{providerUrl}} JMS provider specific provider URL used to configure a connection
     F{{connectionFactoryName}} JMS connection factory to be used in creating JMS connections
+    F{{username}} Username for the JMS connection
+    F{{password}} Password for the JMS connection
     F{{properties}} Additional properties use in initializing the initial context
 }
 public type ConnectionConfiguration {
     string initialContextFactory = "wso2mbInitialContextFactory";
     string providerUrl = "amqp://admin:admin@ballerina/default?brokerlist='tcp://localhost:5672'";
     string connectionFactoryName = "ConnectionFactory";
+    string username;
```

Comment:
We need to define username and password as nullable strings as [1], if not you'll probably get an empty string value here[1] for username and password when the user doesn't configure username and password.

[1].
```
public type ConnectionConfiguration {
    string initialContextFactory = "wso2mbInitialContextFactory";
    string providerUrl = "amqp://admin:admin@ballerina/default?brokerlist='tcp://localhost:5672'";
    string connectionFactoryName = "ConnectionFactory";
    string username?;
    string password?;
    map properties;
};
```

[2]. https://github.com/ballerina-platform/ballerina-lang/pull/9212/files#diff-01450798131cea146644def4f81988c7R116

Reply:
@VijithaEkanayake, @shafreenAnfar Shall we merge this PR to separate branch and then fix said improvements. After that, we can merge this to master.

Reply:
I actually tried putting in the question marks, for the optional elements, but then the build failed with the following statement:
error: ballerina/jms/connection.bal:57:20: extraneous input '?'
error: ballerina/jms/connection.bal:58:20: extraneous input '?'

How would I get around this issue?

Reply:
@JensWalter Could you try as below. "?" should be associated with the type not with the variable. I've accidentally added it to the variable name in the sample. Apologies for the confusion made.

```
public type ConnectionConfiguration {
    string initialContextFactory = "wso2mbInitialContextFactory";
    string providerUrl = "amqp://admin:admin@ballerina/default?brokerlist='tcp://localhost:5672'";
    string connectionFactoryName = "ConnectionFactory";
    string? username;
    string? password;
    map properties;
};
```

Reply:
yes, that seems to work much better.
Now that I changed the definition of the type, does that have any influence on how ballerina maps values into the java space. As of right now, I'm only checking for 'null', should I also check for empty values?

current code section:
```            
String username = connectionConfig.getStringField(Constants.ALIAS_USERNAME);
String password = connectionConfig.getStringField(Constants.ALIAS_PASSWORD);
if (username != null && password != null) {
     return connectionFactory.createConnection(username, password);
} else {
     return connectionFactory.createConnection();
}
```

Reply:
Ballerina never assigns a null value for a data type unless the user specifically defines the type should be nullable. The Ballerina has set of default values for each of the data types. for example, default of string is empty string default of int is 0, boolean is false etc... Currently, we don't have a proper document to refer those values and I created issue[1] to track the documentation.

In here, As we define the username and password as nullable we don't need to check for the empty string. Checking the empty string is dependent on the use case. for example, If any case user has passed empty values for these parameters we might need to check the empty strings as well, assuming that empty string are not allowed to pass as parameter values.

[1]. https://github.com/ballerina-platform/ballerina-lang/issues/9226

Reply:
I made the `username` and `password` optional and changed the cheking for empty values from 'null' to `isEmpty()`